### PR TITLE
Updated civetweb to v1.8

### PR DESCRIPTION
**Pull request by pritha-srivastava**
Updated civetweb version to v1.8.
Split out commits from Marcus's civetweb branch, and applied the same here.

### CODE REVIEW

**Comment by pritha-srivastava** [comment]
The civetweb repo is based off https://github.com/pritha-srivastava/civetweb:wip-civetweb-1.8.
One commit related to 'fix shadowed automatic variable: fstore.' in Marcus's branch wip-easy1 has been removed since it doesn't apply anymore.

**Comment by liewegas** [comment]
this needs to be rebased on current master (and remove the autotools changes)

File: CMakeLists.txt [hunk 1]
```diff
@@ -366,6 +366,12 @@ option(DEBUG_GATHER "C_Gather debugging is enabled" ON)
 option(HAVE_LIBZFS "LibZFS is enabled" OFF)
 option(ENABLE_COVERAGE "Coverage is enabled" OFF)
 option(PG_DEBUG_REFS "PG Ref debugging is enabled" OFF)
+# we want to include civetweb.h as "civetweb/civetweb.h".  Make it so.
+execute_process(
+COMMAND rm -f "${CMAKE_BINARY_DIR}/src/include/civetweb"
+COMMAND mkdir -p "${CMAKE_BINARY_DIR}/src/include"
+COMMAND ln -s "${CMAKE_SOURCE_DIR}/src/civetweb/include"
+        "${CMAKE_BINARY_DIR}/src/include/civetweb")
```
Comment (cbodley) [inline CMakeLists.txt]: i'm confused by this symlink stuff - it says we want to include as `"civetweb/civetweb.h"`, but the pr doesn't contain any such changes

**Comment by yehudasa** [comment]
merged in a separate PR

**Pull request closed without merging**